src/chain.h

```diff
@@ -212,7 +215,8 @@ class CBlockIndex
     {
     }
 
-    FlatFilePos GetBlockPos() const {
+    FlatFilePos GetBlockPos() const EXCLUSIVE_LOCKS_REQUIRED(cs_main)
```

Comment:
This is annotating the function, but not the underlying data member that needs the annotation.

Not sure how involved it will be to annotate everything properly here. Also, given that cs_main is the validation lock, we should (eventually/ideally) have a separate node_storage lock. Though, this again requires a major validation/node_storage rewrite.

However an argument against adding the lock annotation to the data member could be that `chain` is supposed to be a data-structure-only primitive, like `transaction` or `block`. In that case only annotating the getter and setter functions (and making them standalone) would seem appropriate.

Reply:
Having a quick look at annotating `nStatus` to assess.

Reply:
Appended an exploratory small commit guarding `CBlockIndex::nStatus` by `cs_main`.